common_features.mk

```diff
@@ -254,20 +254,32 @@ QUANTUM_SRC:= \
     $(QUANTUM_DIR)/keymap_common.c \
     $(QUANTUM_DIR)/keycode_config.c
 
-ifeq ($(strip $(SPLIT_KEYBOARD)), yes)
-    ifneq ($(strip $(CUSTOM_MATRIX)), yes)
-       QUANTUM_SRC += $(QUANTUM_DIR)/split_common/matrix.c
-       # Do not use $(QUANTUM_DIR)/matrix.c.
-       CUSTOM_MATRIX=yes
+# Include the standard or split matrix code if needed
+ifneq ($(strip $(CUSTOM_MATRIX)), yes)
```

Comment:
Was there a specific reason for reverting the `CUSTOM_MATRIX` code?

Reply:
Locality, basically. The original code had a single block that was only concerned with including or excluding standard matrix code. Whether the matrix.c was split or non-split was a minor detail and was checked in the subordinate clause.

The change swapped the primary and subordinate clauses. Instead of dealing with the question of "is the standard matrix code required", it was trying to cram everything to do with split keyboards into one block and everything else in another.

That split the original matrix code into two separate blocks that then needed to communicate with each other by actively changing the meaning of CUSTOM_MATRIX halfway through (at the start of the file, it's "matrix code is being provided elsewhere", and after the first block it's "either custom matrix code is being provided, OR the standard split matrix code.") It took some effort to confirm that there was no way of accidentally including both standard implementations, as it was no longer a simple "if/else".